### PR TITLE
Refactor command dicts to actual payload

### PR DESCRIPTION
**Pull request by itsthejoker**
## Description:

@TimJentzsch made a comment a while ago about how he wished the commands took a standardized object that could take advantage of type hinting and code suggestions instead of just having everything buried inside an arbitrary dict. I had an idea, and now this is a thing.

## Checklist:

- [x] Code Quality
- [x] Pep-8
- [ ] Tests (if applicable)
- [x] Success Criteria Met
- [x] Inline Documentation
- [ ] Wiki Documentation (if applicable)

### CODE REVIEW

File: bubbles/message_utils.py [hunk 1]
```diff
@@ -1,19 +1,74 @@
-from typing import Any, Dict, Union
+from __future__ import annotations
 
+from typing import Any, Dict, Union, Callable, Optional, TYPE_CHECKING
+from slack_sdk.web.client import WebClient
 
-class MessageUtils:
-    def __init__(self, client, payload):
+if TYPE_CHECKING:
+    from bubbles.plugins import PluginManager
```
Comment (TimJentzsch) [inline bubbles/message_utils.py:7]: What is this for?

File: bubbles/message_utils.py [hunk 2]
```diff
@@ -1,19 +1,74 @@
...
+
+class Payload:
+    """Payload class for everything a command needs."""
+    def __init__(
+            self,
+            client: WebClient=None,
+            slack_payload: dict=None,
+            say: Callable=None,
+            context: dict=None,
+            meta: PluginManager=None,
```
Comment (TimJentzsch) [inline bubbles/message_utils.py]: I think this is not correctly formatted (but I'm not sure), did you run black?

File: bubbles/message_utils.py [hunk 3]
```diff
@@ -1,19 +1,74 @@
...
+    ):
         self.client = client
-        self.payload = payload
+        self._payload = slack_payload
```
Comment (TimJentzsch) [inline bubbles/message_utils.py]: Let's call it `self.slack_payload` to make the distinction clearer

**Review by TimJentzsch — COMMENTED** [review]
Much nicer than before, well done!
I just have some smaller comments

File: bubbles/message_utils.py (hunk 1 above)
Comment (itsthejoker) [reply bubbles/message_utils.py:7]: Type hinting without a circular dependency error. It lets the class be imported for the IDE to parse, but won't get evaluated during runtime.

File: bubbles/message_utils.py (hunk 2 above)
Comment (itsthejoker) [reply bubbles/message_utils.py]: Oops I probably didn't.

File: bubbles/message_utils.py (hunk 3 above)
Comment (itsthejoker) [reply bubbles/message_utils.py]: I want this to be a "private" variable - the idea is that if you need something out of it, write a function that can be found by your editor. How about _slack_payload?

Comment (TimJentzsch) [reply bubbles/message_utils.py]: That works for me!

**Pull request merged**